docs/architecture/node-high-level.md

```diff
@@ -1 +1,117 @@
+
+The service provider sends to the consumer all required information for establishing VPN session.
+
+## Service consumer
+
+The same `myst` binary could be used as a service consumer. The service consumer should have a similar registered identity as a service provider to be able to communicate with it.
+
+Once consumer identity created and registered in the network, a user can request discovery service for the list of available service proposals. The results of this request can be filtered through the discovery API.
+
+Service consumer chooses the service proposal that fit his requirements and starts communication dialog to negotiate session creation.
+
+Service consumer gets from the provider information required for establishing VPN session. It initializes the VPN session creation process.
+
+When the VPN session established, the service consumer can consume provided services as he needs. And service consumer can close the session any time he wants.
+
+## User identity
+
+There are interconnected software agents (we call one instance of this an identity agent) representing digital identities. Each identity agent acts on behalf of a person controlling the identity.
+
+This software agent is a functional part of the application (a Mysterium network node) used to connect to the Mysterium network to provide or consume VPN services.
+
+The identity is created by generating public and private keys. The identity is identified by a unique identifier derived from a public key by using the last 20 bytes of a keccak256 hash of the public key.
+
+This identity can be made publicly known to other network users by announcing its existence through invoking an identity registration smart contract on the Ethereum blockchain.
+
+After the identity contract is successfully executed the public key of identity is appended to the Ethereum blockchain by miners. At this point, the appended identity becomes a Registered Identity.
+
+Nodes can use the Registered Identities database to lookup public keys associated with other identities. This database is used by nodes to check if the communications received from other nodes comes from validly registered identities and are properly signed.
+
+## Service discovery
+
+Service providers willing to provide VPN services can announce their services to the network.
+
+Every client is free to choose different services from the same or multiple providers on the
+Mysterium network. A consumer of VPN services uses Discovery API to search for a service adhering to their requirements:
+* service type;
+* locations of where service is provided, where your traffic will appear from;
+* metering units;
+* maximum price per metered unit and committed bandwidth.
+
+To announce the service, the provider's node prepares a service proposal.
+
+### Service proposal
+
+A proposal encodes the proposal format version, provider description, qualitative service definition, and a list of methods to reaching the provider's node.
```

Comment:
A proposal encodes the proposal format version, provider description, qualitative service definition, and a list of methods for reaching the provider's node.